pkg/devfile/detect.go

```diff
@@ -196,31 +202,28 @@ func AnalyzePath(a Alizer, localpath, context, devfileRegistryURL string, devfil
 
 // SearchForDockerfile searches for a Dockerfile from a devfile image component and
 // returns the dockerfile uri
-func SearchForDockerfile(devfile []byte) (string, error) {
-	var dockerfile string
-
+func SearchForDockerfile(devfile []byte) (*v1alpha2.DockerfileImage, error) {
 	devfileData, err := ParseDevfileModel(string(devfile))
 	if err != nil {
-		return "", err
+		return nil, err
 	}
 	devfileImageComponents, err := devfileData.GetComponents(common.DevfileOptions{
 		ComponentOptions: common.ComponentOptions{
 			ComponentType: v1alpha2.ImageComponentType,
 		},
 	})
 	if err != nil {
-		return "", err
+		return nil, err
 	}
 
 	for _, component := range devfileImageComponents {
 		// Only check for the Dockerfile Uri at this point, in later stages we need to account for Dockerfile from Git & the Registry
 		if component.Image != nil && component.Image.Dockerfile != nil && component.Image.Dockerfile.DockerfileSrc.Uri != "" {
-			dockerfile = component.Image.Dockerfile.DockerfileSrc.Uri
-			break
+			return component.Image.Dockerfile, nil
 		}
 	}
 
-	return dockerfile, nil
+	return nil, nil
```

Comment:
you can return empty object, so you will not have issue with nil pointer -> `return &v1alpha2.DockerfileImage{}, nil`